### PR TITLE
Make `ship serve local` attempt to serve everything

### PR DESCRIPTION
**Pull request by eob**
Running `ships serve local` will now attempt to load up every option by default. The goal is to have a single command to suggest to people that always works: `ship serve local` 

image

## Behaviors:

* Logging is limited to warning and errors in the console
* A `logs/shiplog-datetime.log` file is used for all other logging messages
* A repl, local server, ngrok server, and web UI are all started simuiltaneously
* Options to suppress any of those startups are provided as command line args

## Limitation:

* The REPL depends upon the latest python-client code which adds a `/prompt` method to ALL AgentService objects. It will not work with older AgentService objects.
  * Suggested followup: A second PR catches the `/prompt` 404 and tries `/answer`, remembering that worked if so

## Assumption:

This depends upon the PR to be merged with `ship server local` -> `ship run local`.

## How to test

Just use `pip install -e path/to/steamship-python` and then `.venv/bin/ship serve local` inside an agent project.

### CODE REVIEW

**Review by douglas-reid — COMMENTED** [review]
nits and questions.

File: src/steamship/cli/cli.py [hunk 1]
```diff
@@ -76,53 +77,205 @@ def ships():
...
+    try:
+        config = Configuration()
+        web_base = config.web_base
+    except BaseException:
+        click.secho("⚠️ Web UI:  Unable to find Steamship Configuration. Please:")
+        click.secho(
+            "   - Run `ship login` to make sure you have Steamship credentials in your environment."
+        )
+
+    web_url = f"{web_base}/debug?endpoint={base_url}/prompt"
```
[the remark this answers is not in the document]
Comment (eob) [reply src/steamship/cli/cli.py:186]: Fixed!

File: src/steamship/cli/cli.py [hunk 2]
```diff
@@ -76,53 +77,205 @@ def ships():
...
+    no_repl: Optional[bool] = False,
+    no_ui: Optional[bool] = False,
+    config: Optional[str] = None,
+    workspace: Optional[str] = None,
+):
+    """Serve the invocable on localhost. Useful for debugging locally."""
+    dev_logging_handler = DevelopmentLoggingHandler.init_and_take_root()
+
+    initialize()
+    click.secho("Running your project..\n")
```
[the remark this answers is not in the document]
Comment (eob) [reply src/steamship/cli/cli.py:205]: Fixed!

File: src/steamship/cli/cli.py [hunk 3]
```diff
@@ -76,53 +77,205 @@ def ships():
...
+        )
+
+    web_url = f"{web_base}/debug?endpoint={base_url}/prompt"
+    logging.info(f"Web interface url is: {web_url}")
+
+    return web_url
+
+
+def serve_local(
+    port: int = 8080,
```
[the remark this answers is not in the document]
Comment (eob) [reply src/steamship/cli/cli.py:193]: fixed!

File: src/steamship/cli/cli.py [hunk 4]
```diff
@@ -76,53 +77,205 @@ def ships():
...
-    "--api_key",
-    "-k",
-    type=str,
-    default=None,
-    help="API Key to hard-code for hosting.",
-)
-@click.option(
-    "--ngrok",
+    "--no_ngrok",
     "-n",
```
[the remark this answers is not in the document]
Comment (eob) [reply src/steamship/cli/cli.py:261]: - Switched to dashcase
- Removed shorthands

Also: I feel like these are kind of weird options in general; I'm OK removing them if folks feel that's better. I added them as an insurance policy: if there's something our testing didn't catch (e.g. with ngrok) then having a way to suppress it feels like a useful last resort when helping a user.

File: src/steamship/agents/service/agent_service.py [hunk 5]
```diff
@@ -87,9 +87,70 @@ def run_agent(self, agent: Agent, context: AgentContext):
             logging.info(f"Emitting via function: {func.__name__}")
             func(action.output, context.metadata)
 
+    def _prompt(self, prompt: str, **kwargs) -> List[Block]:
```
Comment (douglas-reid) [inline src/steamship/agents/service/agent_service.py:90]: what does this do that the other prompt doesn't?

Comment (eob) [reply src/steamship/agents/service/agent_service.py]: REmoved.

**Review by douglas-reid — DISMISSED** [review]

**Review by EniasCailliau — APPROVED** [review]
LGTM

**Pull request merged**